core/app/workers/workarea/generate_sitemaps.rb

```diff
@@ -0,0 +1,71 @@
+module Workarea
+  class GenerateSitemaps
+    include Sidekiq::Worker
+    include Storefront::Engine.routes.url_helpers
+
+    TMP_DIR = Rails.root.join('tmp', 'sitemaps')
```

Comment:
It'd be safer (for handling concurrency and whatnot) to not have this be a constant (unique per instance), and to use a built in Ruby temp dir which can give you a unique dir.

Reply:
Agreed. Honestly didn't even look at the code very hard.